sympy/physics/units/quantities.py

```diff
@@ -117,9 +116,15 @@ def _collect_factor_and_dimension(expr):
             factor, dim = Quantity._collect_factor_and_dimension(expr.base)
             return factor ** expr.exp, dim ** expr.exp
         elif isinstance(expr, Add):
-            raise NotImplementedError
+            factor, dim = Quantity._collect_factor_and_dimension(expr.args[0])
```

Comment:
@asmeurer I have also followed the example in https://github.com/sympy/sympy/pull/12936/files#diff-6790d0f6aee4060f95bd1eeca4dd74d4R110